subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/tasks/AbstractNativeCompileTask.java

```diff
@@ -96,6 +98,19 @@ public String call() throws Exception {
                 return NativeToolChainInternal.Identifier.identify(toolChain, targetPlatform);
             }
         });
+        getOutputs().cacheIf("Header dependency analysis provided", new Spec<Task>() {
+            @Override
+            public boolean isSatisfiedBy(Task element) {
+                AbstractNativeCompileTask compileTask = (AbstractNativeCompileTask) element;
+                return compileTask.getHeaderDependenciesFile().isPresent();
```

Comment:
You can use `AbstractNativeCompileTask.this` instead.

Reply:
fixed.